### ISSUE TITLE
Algorithm passed to Hash is only used if algorithm key is provided before hash key

### ISSUE THREAD

**Issue by InvisibleSmiley** [laminas/laminas-validator#398]
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 2.x

#### Summary

The order of option keys determines whether the given algorithm is used.

#### Current behavior

If the algorithm key is passed after the hash key, the algorithm is silently set to crc32 (default).

#### How to reproduce

```php
// works
new Hash(['algorithm' => 'sha1', 'hash' => 'foo']);
// fails
new Hash(['hash' => 'foo', 'algorithm' => 'sha1']);
```

#### Expected behavior

Order of keys in options array is irrelevant.

**Comment by gsteel** [comment]
FWIW, it's gonna be a while before adoption of validator 3.0 gets any traction, so a fix for this in 2.x will get merged and released if you can provide it 👍

**Comment by gsteel** [comment]
Fixed in #399
